src/disk_io_thread.cpp

```diff
@@ -361,8 +362,9 @@ namespace libtorrent
 		}
 
 		// piece range
-		int range_start = int((p->piece / cont_pieces) * cont_pieces);
-		int range_end = (std::min)(range_start + cont_pieces, p->storage->files()->num_pieces());
+		piece_index_t const range_start((static_cast<int>(p->piece) / cont_pieces) * cont_pieces);
+		piece_index_t const range_end(std::min(static_cast<int>(range_start)
+			+ cont_pieces, p->storage->files()->num_pieces()));
```

Comment:
actually, this expression would become more complex to use ``end_piece()``, since the first argument to ``min()`` would have to be explicitly converted to a ``piece_index_t`` otherwise